applications/GeoMechanicsApplication/custom_utilities/transport_equation_utilities.hpp

```diff
@@ -49,5 +50,24 @@ class GeoTransportEquationUtilities
                RelativePermeability * PermeabilityUpdateFactor * IntegrationCoefficient;
     }
 
+    template <unsigned int TNumNodes>
+    static inline BoundedMatrix<double, TNumNodes, TNumNodes> CalculateCompressibilityMatrix(
+        const Vector&         rNp,
+        double                BiotModulusInverse,
+        double                IntegrationCoefficient,
+        std::optional<double> DtPressureCoefficient = std::nullopt)
+    {
+        return CalculateCompressibilityMatrix(rNp, BiotModulusInverse, IntegrationCoefficient, DtPressureCoefficient);
+    }
+
+    static inline Matrix CalculateCompressibilityMatrix(const Vector& rNp,
+                                                        double        BiotModulusInverse,
+                                                        double        IntegrationCoefficient,
+                                                        std::optional<double> DtPressureCoefficient = std::nullopt)
+    {
+        auto result = -PORE_PRESSURE_SIGN_FACTOR * BiotModulusInverse * outer_prod(rNp, rNp) * IntegrationCoefficient;
+        if (DtPressureCoefficient.has_value()) return result * DtPressureCoefficient.value();
+        else return result;
```

Comment:
I would simplify this to:
```suggestion
        if (DtPressureCoefficient) result *= DtPressureCoefficient.value();
        return result;
```
It is simpler because a `std::optional` object can be implicitly tested (no need to call `has_value()`), there is no `else` clause needed and we can have a single exit only (rather than having two).

Reply:
Thank you for the suggestion but based on your another suggestion DtPressureCoefficient has been removed from this function.